lib/vtls/mbedtls.c

```diff
@@ -802,15 +882,7 @@ mbed_connect_step1(struct Curl_cfilter *cf, struct Curl_easy *data)
     failf(data, "mbedTLS: ssl_config failed");
     return CURLE_SSL_CONNECT_ERROR;
   }
-#ifdef TLS13_SUPPORT
-  if(!verifypeer) {
-    /* Default verify behaviour changed in mbedtls v3.6.0 with TLS v1.3.
-     * On 1.3 connections, the handshake fails by default without trust
-     * anchors. We override this questionable change by installing our
-     * own verify callback that clears all errors. */
-    mbedtls_ssl_conf_verify(&backend->config, mbed_no_verify, cf);
-  }
-#endif
+  mbedtls_ssl_conf_verify(&backend->config, mbed_verify, &cb_data);
```

Comment:
This change seems unrelated and possibly even wrong?

Reply:
This is related and correct. The `verifypeer` check is now done inside the callback function. If `verifypeer` is false then `mbed_verify()` will clear the flags the same way that `mbed_no_verify` did.

We log the cert info inside the callback function now; otherwise we can't log it if cert validation fails.